### ISSUE TITLE
UI Sign-in button broken for OIDC auth method with Okta

### ISSUE THREAD

**Issue by NightmareCinema** [hashicorp/vault#10250]
**Describe the bug**
UI sign-in button doesn't do anything when using a configured OIDC auth method (tested with Okta)

**To Reproduce**
1. Configure OIDC auth method using Okta
2. Attempt to login via the UI with said auth method
3. Nothing happens (when capturing network request logs in Chrome only changes to the **Role** textbox generate requests, but not actual clicks on the sign-in button). 

**Expected behavior**
Logging in successfully.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.5.4
* Vault CLI Version (retrieve with `vault version`): 1.5.5
* Server Operating System/Architecture: Kubernetes architecture (Official Helm chart version 0.8.0)

Vault server configuration file(s):

```hcl
ui = true
            
listener "tcp" {
    address = "[::]:8200"
    cluster_address = "[::]:8201"
    tls_cert_file = "/etc/pki/tls.crt"
    tls_key_file  = "/etc/pki/tls.key"
}

storage "gcs" {
    bucket = "my-gcs-bucket"
    ha_enabled = "true"
}

service_registration "kubernetes" {}
```

**Additional context**
Using the CLI to log in with said auth method works perfectly using the following command:
`vault login -method=oidc -path=my_okta_oidc role=my_role`

**Comment by jeffsanicola** [comment]
Same issue with Azure. Running Vault 1.5.4

**Comment by tvoran** [comment]
Hi folks, I'm not able to reproduce this issue, so I wonder if you've double-checked the redirect_urls in your oidc config, and also if there's anything being logged on the vault side? You also may want to test a different web browser too, in case there's an extension blocking something.

**Comment by jeffsanicola** [comment]
Hey @tvoran - good call - I had an extra / in my URL.  Got it working now!

**Comment by tvoran** [comment]
@NightmareCinema So in trying this some more, I found that if the default_role was set to a non-existent vault role, then I get the behavior you describe. Wondering if that is what's happening in your case?

**Comment by asafha-etorox** [comment]
> Hey @tvoran - good call - I had an extra / in my URL. Got it working now!

@jeffsanicola Can you please share your before & after configuration?

**Comment by NightmareCinema** [comment]
> @NightmareCinema So in trying this some more, I found that if the default_role was set to a non-existent vault role, then I get the behavior you describe. Wondering if that is what's happening in your case?

@tvoran I left that empty actually.
Still, it should be ignored when I specifically define one at login time right?

**Comment by NightmareCinema** [comment]
> Hey @tvoran - good call - I had an extra / in my URL. Got it working now!

@jeffsanicola Can you please share your before & after configuration?

**Comment by NightmareCinema** [comment]
> Hi folks, I'm not able to reproduce this issue, so I wonder if you've double-checked the redirect_urls in your oidc config, and also if there's anything being logged on the vault side? You also may want to test a different web browser too, in case there's an extension blocking something.

That was it. I had a mismatch in my **allowed redirect urls** caused by my port number. Now everything is truly identical and works smoothly! Many thanks!

**Comment by PeterSR** [comment]
In my case I too had forgot to a add a role (apparently something that needs to be done via the CLI despite everything else being done via the UI). The way I discovered it was by enabling the browser dev tool and via network requests. When selecting OIDC in the dropdown it would make a request that fails with status code 400 and an error message about a missing role. 

After the role was added, it still did not work until I realized that Firefox would block the popup Vault UI was making. 

In my opinion it _is_ a bug that no error message is printed when the role is missing and OIDC is selected.
